src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1393,17 +1398,25 @@ private void displayMessagesAndResult(boolean isSigning) {
     }
 
     private String verifyWithWeak(String alg, Set<CryptoPrimitive> primitiveSet,
-        boolean tsa, JarConstraintsParameters jcp) {
+        boolean tsa, JarConstraintsParameters jcp, AlgorithmParameters algParams) {
 
         try {
             JAR_DISABLED_CHECK.permits(alg, jcp, false);
         } catch (CertPathValidatorException e) {
             disabledAlgFound = true;
             return String.format(rb.getString("with.disabled"), alg);
         }
+        if (algParams != null) {
+            try {
+                JAR_DISABLED_CHECK.permits(algParams, jcp);
+            } catch (CertPathValidatorException e) {
+                disabledAlgFound = true;
+                return String.format(rb.getString("with.disabled"), algParams);
```

Comment:
The return value should still be the algorithm name with an optional "weak" (or "disabled") tag, something like "RSASSA-PSS (weak)". However, the updated code is now returning the string format of `algParams`, which will be "PSSParameter... (weak)".

I'm not sure how to fix this nicely. Certainly you want to show the user why it is weak so the weak part should be displayed. A verbose solution could be "RSSSSA-PSS using PSSParameterSpec(SHA-256) (weak)". Hopefully the `toString()` output of `PSSParameterSpec` is not too ugly.

Reply:
I add "RSSSSA-PSS using “ to the `-verbose` output as suggested, and keep the remaining output as the parameters for the RSASSA-PSS contain hashAlgorithm and maskGenAlgorithm that could be disabled or weak. At the same time, strip off the saltLength and trailerField display.

Reply:
What does it look like now? Also, you might need to create a mapping in `Resources.java` because "using" should only be shown when system language is English.

Reply:
Also, what if it's another algorithm using another type of parameters? You cannot hardcode "RSASSA-PSS" and take it for granted that there is a "]" inside the string format of the parameter and it's the end of the weak part.

Reply:
Made change to add "RSASSA-PSS using” before its parameter output when the signature algorithm is RSASSA-PSS. Also, keep the parameter string without doing further parsing and stripping off based on the "]".